internal/tlscert/tls_cert_validator.go

```diff
@@ -142,6 +173,18 @@ func parsePrivateKey(keyPEM []byte) error {
 	return nil
 }
 
+func validateCA(cert *x509.Certificate, ca *x509.Certificate) error {
+	if !ca.IsCA {
+		return ErrCertIsNotCA
+	}
+
+	if cert.Issuer.String() != ca.Subject.String() {
```

Comment:
This check might be wrong in some situations. This CA is used to sign the server certificate, but the configured certificate of a pipeline is the client certificate. Both certificates can be signed by the same CA, but this is not mandatory.

Reply:
I think I understand now. My intuition was that the CA is always the one that signs the client certificate.